add pipeline client configuration introduction (#15459)
Azure/azure-sdk-for-python: sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

response = client.get_foo_properties()
```

### Pipeline client configurations

| Parameters | Description |
| --- | --- |

> | pipeline | While `PipelineClient` will create a default pipeline. User can opt in to use their own pipeline by passing in a `Pipeline` object. If it is specified, the rest configurations will be ignored.  |

Reviewer: nit: can you add `` aroud the parameters, i.e. `pipeline`

> | pipeline | While `PipelineClient` will create a default pipeline. User can opt in to use their own pipeline by passing in a `Pipeline` object. If it is specified, the rest configurations will be ignored.  |

Reviewer: ```suggestion
| `pipeline` | While `PipelineClient` will create a default pipeline, users can opt to use their own pipeline by passing in a `Pipeline` object. If passed in, the other configurations will be ignored.  |
```

Reviewer: This structure can be applied to the other sections as well

> | `config` | While `PipelineClient` will create a default `Configuration`, users can opt to use their own configuration by passing in a `Configuration` object. If passed in, it will be used to create `Pipeline` object. |

Reviewer: tiny nit: "it will be used to create a `Pipeline` object."

> | `transport` | While `PipelineClient` will create a default `RequestsTransport`, users can opt to use their own transport by passing in a `RequestsTransport` object. If it is omitted, `PipelineClient` will honor other transport customization described in [Transport](#transport). |

Reviewer: tiny nit:  "If it is omitted, `PipelineClient` will honor other the described transport customizations"
